### PR TITLE
Scheduled report

### PR DESCRIPTION
**Pull request by cdwilhelm**
### Description

A new policy for scheduled reports

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/scheduled_reports/scheduled_report.pt [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+      var current_month_total = 0.00
+      for (var i = 0; i < current_month_costs['rows'].length; i++) {
+        var row =  current_month_costs['rows'][i]
+        current_month_total += row['metrics'][metric]
+      }
+
+      var weekly_avg = 0
+      if (current_week_costs['rows'].length > 0){
+        //weekly_avg = current_week_total / current_week_costs['rows'].length
```
Comment (douglaswth) [inline cost/scheduled_reports/scheduled_report.pt]: Should probably remove commented out code.

File: cost/scheduled_reports/scheduled_report.pt [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+        var days_this_week = current_week_costs['rows'].length
+        weekly_avg = (current_week_total + previous_week_total) / (7 + days_this_week)
+      }
+
+      var monthly_avg = 0
+      if (current_month_costs['rows'].length > 0){
+        //monthly_avg = current_month_total / current_month_costs['rows'].length
```
Comment (douglaswth) [inline cost/scheduled_reports/scheduled_report.pt]: Same as above.

File: cost/scheduled_reports/scheduled_report.pt [hunk 3]
```diff
@@ -0,0 +1,306 @@
+name "Scheduled Report"
+rs_pt_ver 20180301
+type "policy"
+short_description "This policy allows you to set up scheduled reports that will provide summaries of cloud cost across all resources in the billing centers you specify, delivered to any email addresses you specify. The policy will report the following:
```
Comment (douglaswth) [inline cost/scheduled_reports/scheduled_report.pt]: Should the short description be this long and the long description be so short?

File: cost/scheduled_reports/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+The following policy actions are taken on any resources found to be out of compliance.
+
+- Send an email report
+
+## Required Permissions
+
+This policy requires permissions to access RightScale resources (optima).  Before applying this policy add the following roles to the user applying the policy.  The roles should be applied to all Accounts where the policy will run or the Organization. For more information on modifying roles visit the [Governance Docs](https://docs.rightscale.com/cm/ref/user_roles.html)
+
+- Optima - billing_center_viewer
```
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply cost/scheduled_reports/README.md]: the optima.cost resources isn't available to the policy template yet.

File: cost/scheduled_reports/scheduled_report.pt [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+parameter "param_cost_metric" do
+  type "string"
+  label "Cost Metric"
+  allowed_values "Unamortized Unblended","Amortized Unblended","Unamortized Blended","Amortized Blended"
+  default "Unamortized Unblended"
+  description "Select the cost metric for your report.  See the README file for more details"
+end
+
+##  the below params are used to pass values to the `costs` datasources
+##  these are temperay until we can pass literal values to the datasource parameters
```
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply cost/scheduled_reports/scheduled_report.pt]: fixed

File: cost/scheduled_reports/scheduled_report.pt [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+parameter "param_weekly_range" do
+  type "string"
+  label "param_weekly_range"
+  description "DO NOT CHANGE"
+  no_echo true
+  default  "daily"
```
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply cost/scheduled_reports/scheduled_report.pt]: changed.  good catch.

File: cost/scheduled_reports/scheduled_report.pt [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+  description "Select the cost metric for your report.  See the README file for more details"
+end
+
+##  the below params are used to pass values to the `costs` datasources
+##  these are temperay until we can pass literal values to the datasource parameters
+parameter "param_weekly_range" do
+  type "string"
+  label "param_weekly_range"
+  description "DO NOT CHANGE"
+  no_echo true
```
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply cost/scheduled_reports/scheduled_report.pt]: agreed.  removed `no_echo`

File: cost/scheduled_reports/scheduled_report.pt [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+  default  "daily"
+end
+
+parameter "param_monthly_range" do
+  type "string"
+  label "param_montly_range"
```
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply cost/scheduled_reports/scheduled_report.pt]: fixed

File: cost/scheduled_reports/scheduled_report.pt (hunk 1 above)
Comment (cdwilhelm) [reply cost/scheduled_reports/scheduled_report.pt]: removed

File: cost/scheduled_reports/scheduled_report.pt (hunk 3 above)
Comment (cdwilhelm) [reply cost/scheduled_reports/scheduled_report.pt]: changed

**Pull request merged**